cmake/TorchConfig.cmake.in

```diff
@@ -38,22 +38,40 @@ endif()
 # Library dependencies.
 if (@BUILD_SHARED_LIBS@)
   find_package(Caffe2 REQUIRED PATHS ${CMAKE_CURRENT_LIST_DIR}/../Caffe2)
-endif()
-
-if (NOT ANDROID)
-  find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  set(TORCH_LIBRARIES torch ${Caffe2_MAIN_LIBS})
 else()
-  find_library(TORCH_LIBRARY NO_CMAKE_FIND_ROOT_PATH torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
```

Comment:
These "NO_CMAKE_FIND_ROOT_PATH" lines were added by me: https://github.com/pytorch/pytorch/pull/19762/files#r278829259 - I simpily copied them from @t-vi 's original PR.
Now look at these lines again I don't think they are actually useful any more:
1) when building with new NDK r19 it won't use system CMAKE_FIND_ROOT_PATH anyway: https://github.com/android/ndk/issues/912
2) The mobile build script will install the library into a different location than host build - so even if we need differentiate between mobile v.s. host build this can be done at build time with "if (NOT @INTERN_BUILD_MOBILE@)" instead of "if (NOT ANDROID)". I cannot think of a scenario where the generated cmake file can be used by both host and mobile toolchains.

Reply:
Yeah, maybe NDK r19c doesn't need it anymore. It used to be a problem when I wrote the initial patch.